server/testserver.go

```diff
@@ -28,8 +28,19 @@ import (
 	"github.com/cockroachdb/cockroach/storage/engine"
 	"github.com/cockroachdb/cockroach/util"
 	"github.com/cockroachdb/cockroach/util/hlc"
+	"github.com/cockroachdb/cockroach/util/log"
 )
 
+// StartTestServer starts a in-memory test server.
+func StartTestServer() *TestServer {
```

Comment:
That's fine...we have other examples of this as well. I think the value of making these exported test functions explicitly for testing outweighs having to check here whether the supplied `testing.TB` param is nil.